README.horizontal-scaling.md

```diff
@@ -0,0 +1,160 @@
+The (currently non-configurable) time ranges for *offpeak*, *peak* and *weekend* are:
+
+* peak: M-F, 7am-11pm
+* offpeack: M-F, 11pm-7am
+* weekend: 12am Sat - 12pm Sun
+
+### Auto-scaling
+
+There are two types of auto scaling mechanisms available: `LayerLoad` and `HighLoadJobs`.
+Only the `LayerLoad` type should be used at this time as the `HighLoadJobs` mechanism
+makes use of some unreliable data from the Matterhorn statistics API.
+
+#### LayerLoad
+
+This method scales workers up/down based on the value of an Opsworks
+layer's cloudwatch metric. It compares the metric's reported values against a couple
+of threshold settings and scales up or down accordingly.
+
+Example config:
+
+    {
+      "stack": {
+        "chef": {
+          "custom_json": {
+            "moscaler": {
+              "moscaler_strategy": "auto",
+              "autoscale_type": "LayerLoad"
+              "autoscale_up_increment" => 2,
+              "autoscale_down_increment" => 2,
+              "autoscale_up_threshold" => 6.0,
+              "autoscale_down_threshold" => 10.0,
+              "autoscale_cloudwatch_metric" => "load_5"
+            },
+            ...
+          }
+        }
+      }
+    }
+    
+This configuration would result in `moscaler` scaling up two workers if the `load_5` metric
+for the workers layer reports a value equal to or higher than 6.0 at 1 minute intervals
+for 5 consecutive minutes. It would attempt to scale down two workers if that same metric 
+was less than 10.0 for 5 consectutive minutes. (Note: the 5min sample size and 1min intervals 
+are not currently configurable but could be made so if needed.)
+
+#### HighLoadJobs
+
+Don't use this one. :)
+
+Unfortunately the logic of the `HighLoadJobs` method uses some unreliable data from the Matterhorn
+API. It's in there for reference in case we ever make Matterhorn's job dispatching behave in a way
+we'd like.
+
+### Cluster config settings
+
+The defaults for all moscaler settings are defined in a `mh-opsworks-recipes` helper function:
+
+    def get_moscaler_info
+      {
+          'moscaler_strategy' => 'disabled', 
+          'moscaler_release' => 'v0.4.0',
+          'moscaler_debug' => false,
+          'offpeak_instances' => 2,
+          'peak_instances' => 10,
+          'weekend_instances' => 1,
+          'cron_interval' => '*/5',
+          'autoscale_up_increment' => 1,
+          'autoscale_down_increment' => 1,
+          'autoscale_up_threshold' => 6.0,
+          'autoscale_down_threshold' => 6.0,
+          'autoscale_type' => 'LayerLoad',
+          'autoscale_cloudwatch_metric' => 'load_5',
+          'min_workers' => 1,
+          'idle_uptime_threshold' => 50,
+          'autoscale_up_pause_interval' => 600
+        }.merge(node.fetch(:moscaler, {}))
+    end
+
+* `moscaler_strategy`: either "time" or "auto". Set to empty string or "disabled" to disable moscaler. 
+  The recipe will still install the `mo-scaler` software but no cron entries will be created.
```

Comment:
If there is no "moscaler" block (or an empty block, or an empty `moscaler_strategy` value in the block) in the cluster config the recipe still runs and moscaler is installed but no cron entries are created and it never gets executed.